src/NServiceBus.Core/Transports/TransportSeam.cs

```diff
@@ -42,10 +46,17 @@ public static TransportSeam Create(Settings transportSeamSettings, HostingCompon
                 hostingConfiguration.HostInformation.DisplayName, hostingConfiguration.StartupDiagnostics,
                 hostingConfiguration.CriticalError.Raise, hostingConfiguration.ShouldRunInstallers);
 
-            return new TransportSeam(transportDefinition, settings, transportSeamSettings.QueueBindings);
+            var transportSeam = new TransportSeam(transportDefinition, settings, transportSeamSettings.QueueBindings);
+
+            hostingConfiguration.Services.ConfigureComponent(() => transportSeam.TransportInfrastructure.Dispatcher, DependencyLifecycle.SingleInstance);
```

Comment:
this could also be moved to the SendComponent again and subscribe to the `TransportInfrastructureCreated` event to make the registration work.